### PR TITLE
Adding hpfrec

### PR DESCRIPTION
**Pull request by mdekstrand**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mdekstrand** [comment]
@conda-forge/help-python-c, ready for review!

File: recipes/hpfrec/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+  host:
+    - python
+    - cython >=3.0.0
+    - setuptools
+    - wheel
+    - pip
+    - python
+    - numpy >=1.25
```
Comment (xhochy) [inline recipes/hpfrec/meta.yaml:29]: Always pin exactly in host 
```suggestion
    - numpy 1.25
```

File: recipes/hpfrec/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - scipy >=1.11.1
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipes/hpfrec/meta.yaml:33]: Numpy has a run_exports and this should no longer be used.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by mdekstrand** [comment]
@xhochy thanks! I've made those changes, but CI is currently failing due to what look like Conda-Forge infrastructure problems.

**Review by xhochy — APPROVED** [review]

**Comment by mdekstrand** [comment]
@xhochy I also needed to conditionalize the other `numpy`, but this should work I think.

File: recipes/hpfrec/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  skip: true  # [py<310 or win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib('c') }}
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
```
Comment (xhochy) [inline recipes/hpfrec/meta.yaml:21]: Please also put them in host while https://github.com/conda-forge/openmp-feedstock/issues/126 is not clearly resolved.
```suggestion
    - llvm-openmp  # [osx]
    - libgomp      # [linux]
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Small comment, otherwise LGTM

**Comment by mdekstrand** [comment]
@xhochy all done, ready to merge I think!

**Review by xhochy — APPROVED** [review]

**Pull request merged**